cmd/deployment-check/service.go

```diff
@@ -0,0 +1,366 @@
+
+	return createChan
+}
+
+// deleteService deletes the created test service.
+func deleteService() error {
+
+	deleteChan := make(chan error)
+
+	go func() {
+		defer close(deleteChan)
+		for {
+			// Watch that it is gone.
+			watch, err := client.CoreV1().Services(checkNamespace).Watch(metav1.ListOptions{
+				Watch:         true,
+				FieldSelector: "metadata.name=" + checkServiceName,
+				// LabelSelector: defaultLabelKey + "=" + defaultLabelValueBase + strconv.Itoa(int(now.Unix())),
+			})
+			if err != nil {
+				log.Infoln("Error creating watch client:", err)
+				deleteChan <- err
+				return
+			}
+
+			// Watch for a DELETED event.
+			for event := range watch.ResultChan() {
+
+				log.Debugln("Received an event watching for service changes:", event.Type)
+
+				s, ok := event.Object.(*corev1.Service)
+				if !ok {
+					log.Debugln("Got a watch event for a non-service object -- ignoring.")
+					continue
+				}
+
+				// We want an event type of DELETED here.
+				if event.Type == "DELETED" {
+					log.Infoln("Received a", event.Type, "while watching for service with name ["+s.Name+"] to be deleted.")
+					deleteChan <- nil
+					return
+				}
+			}
+
+			watch.Stop()
+		}
+	}()
+
+	log.Infoln("Attempting to delete service in", checkNamespace, "namespace.")
+	if client == nil {
+		err := errors.New("nil kubernetes client")
+		log.Debugln("deleteService:", err)
+		return err
+	}
+
+	// Make a delete options object to delete the service.
+	deletePolicy := metav1.DeletePropagationForeground
+	graceSeconds := int64(1)
+	deleteOpts := metav1.DeleteOptions{
+		GracePeriodSeconds: &graceSeconds,
+		PropagationPolicy:  &deletePolicy,
+	}
+
+	// Delete the service.
+	err := client.CoreV1().Services(checkNamespace).Delete(checkServiceName, &deleteOpts)
+	if err != nil {
+		log.Infoln("Could not delete service:", checkServiceName)
+		log.Infoln("Beginning backoff retry loop to delete service.")
+	}
+
+	return <-deleteChan
+}
+
+// cleanUpOrphanedService cleans up services created from previous checks.
+func cleanUpOrphanedService() error {
+
+	cleanUpChan := make(chan error)
+
+	go func() {
+		defer close(cleanUpChan)
+		for {
+			// Watch that it is gone.
+			watch, err := client.CoreV1().Services(checkNamespace).Watch(metav1.ListOptions{
+				Watch:         true,
+				FieldSelector: "metadata.name=" + checkServiceName,
+				// LabelSelector: defaultLabelKey + "=" + defaultLabelValueBase + strconv.Itoa(int(now.Unix())),
+			})
+			if err != nil {
+				log.Infoln("Error creating watch client.")
+				cleanUpChan <- err
+				return
+			}
+
+			// Watch for a DELETED event.
+			select {
+			case event := <-watch.ResultChan():
+				log.Debugln("Received an event watching for service changes:", event.Type)
+
+				s, ok := event.Object.(*corev1.Service)
+				if !ok {
+					log.Debugln("Got a watch event for a non-service object -- ignoring.")
+					continue
+				}
+
+				// We want an event type of DELETED here.
+				if event.Type == "DELETED" {
```

Comment:
consider moving to a constant based on the `watch` package

Reply:
Changes made in both service.go and deployment.go